Firebase/InstanceID/FIRInstanceIDCheckinStore.m

```diff
@@ -108,8 +108,21 @@ - (void)saveCheckinPreferences:(FIRInstanceIDCheckinPreferences *)preferences
     return;
   }
 
+  // Save all other checkin preferences in a plist
+  NSError *error;
+  if (![self.plist writeDictionary:checkinPlistContents error:&error]) {
```

Comment:
In the previous version of the code this data was not saved to the plist in case if it failed to be stored to the keychain. Is it fine if we keep in the plist now or we should remove it?

Reply:
The order should not matter since whichever fails, it will produce an error that causing the attention here. Previously the order was causing issue.

Reply:
Sorry if I am missing anything. 
It looks like originally the intention was either save the preferences to both the keychain and plist or not to save it anywhere. Now, as far as I can see, the preferences may be saved to the plist, but fail to be stored to the keychain. Is it an inconsistent state?

Reply:
Because the original attempt, it causes this race condition by using a __block boolean to attempt to wait for first callback to finish, which is def not ideal.

I don't think it hurts if we save the plist but not keychain, because if any of it fail, we will get an error which is clearly stating the issue rather than getting into a confused race condition. I don't think this is a big issue, if it's not consistent.

Reply:
Actually it is not true that both keychain and plist are either saved together or not, It is possible keychain was saved successfully and then plist failed. So the only difference is we save plist first.